bundle/edu.gemini.obslog/src/main/scala/edu/gemini/obslog/instruments/GNIRSLogSegment.scala

```diff
@@ -0,0 +1,154 @@
+package edu.gemini.obslog.instruments
+
+import java.util.logging.Logger
+
+import edu.gemini.obslog.config.model.OlLogItem
+import edu.gemini.obslog.core.OlSegmentType
+import edu.gemini.obslog.obslog.{ConfigMap, InstrumentLogSegment, OlLogOptions}
+import edu.gemini.spModel.gemini.gnirs.{GNIRSParams, InstGNIRS}
+import edu.gemini.spModel.obscomp.InstConstants
+
+import GNIRSParams._
+import GNIRSLogSegment._
+
+import scalaz._
+import Scalaz._
+
+class GNIRSLogSegment(logItems: java.util.List[OlLogItem], obsLogOptions: OlLogOptions) extends InstrumentLogSegment(SegmentType, logItems, obsLogOptions) {
+
+  override def decorateObservationData(map: ConfigMap): Unit = Option(map).foreach(m => {
```

Comment:
Obviously we have to leave this method with the given `ConfigMap` modified to work with the Obslog framework as is.  If we're rewriting this though, one idea would be to build up the changes we want to make to it and apply them all at the end instead of mutating as we go.